pgxn/neon/neon_perf_counters.c

```diff
@@ -0,0 +1,261 @@
+/*-------------------------------------------------------------------------
+ *
+ * neon_perf_counters.c
+ *	  Collect statistics about Neon I/O
+ *
+ * Each backend has its own set of counters in shared memory.
+ *
+ *-------------------------------------------------------------------------
+ */
+#include "postgres.h"
+
+#include <math.h>
+
+#include "funcapi.h"
+#include "miscadmin.h"
+#include "storage/proc.h"
+#include "storage/shmem.h"
+#include "utils/builtins.h"
+
+#include "neon_perf_counters.h"
+#include "neon_pgversioncompat.h"
+
+neon_per_backend_counters *neon_per_backend_counters_shared;
+
+Size
+NeonPerfCountersShmemSize(void)
+{
+	Size		size = 0;
+
+	size = add_size(size, mul_size(MaxBackends, sizeof(neon_per_backend_counters)));
+
+	return size;
+}
+
+bool
+NeonPerfCountersShmemInit(void)
+{
+	bool		found;
+
+	neon_per_backend_counters_shared =
+		ShmemInitStruct("Neon perf counters",
+						mul_size(MaxBackends,
+								 sizeof(neon_per_backend_counters)),
+						&found);
+	Assert(found == IsUnderPostmaster);
+	if (!found)
+	{
+		/* shared memory is initialized to zeros, so nothing to do here */
+	}
+}
```

Comment:
this broke the mac OS build ([CI link](https://github.com/neondatabase/neon/actions/runs/11013557581/job/30582187302?pr=9095)):

```
2024-09-24T12:26:48.6843180Z /Users/runner/work/neon/neon//pgxn/neon/neon_perf_counters.c:50:1: error: non-void function does not return a value [-Werror,-Wreturn-type]
2024-09-24T12:26:48.6844290Z /Applications/Xcode_15.4.app/Contents/Developer/usr/bin/make -C ../../../../src/backend generated-headers
2024-09-24T12:26:48.6892290Z }
2024-09-24T12:26:48.6892680Z make: Nothing to be done for `distprep'.
2024-09-24T12:26:48.6905160Z ^
2024-09-24T12:26:48.6905560Z make: Nothing to be done for `generated-header-symlinks'.
2024-09-24T12:26:48.6948200Z 1 error generated.
```

cc @problame @skyzh

Reply:
https://github.com/neondatabase/neon/pull/9128